src/test/java/com/google/sps/SentimentServletTest.java

```diff
@@ -34,23 +35,26 @@
 public final class SentimentServletTest {
 
   private final String VIDEO_ID = "test id";
-  private final Float SCORE = new Float(0.05f);
+  private final Float SCORE_1 = new Float(0.05);
+  private final Float SCORE_2 = new Float(-0.1);
+  private final Float SCORE_3 = new Float(0.8);
   private StringWriter stringWriter;
   private PrintWriter writer;
   private DecimalFormat df = new DecimalFormat("#.##");
 
   @Mock CommentService commentServiceMock;
   @Mock Caption captionServiceMock;
+  @Mock SentimentAnalysis sentimentAnalysisMock;
   @Mock(answer = Answers.RETURNS_DEEP_STUBS) HttpServletRequest requestMock;
-  @Mock(answer = Answers.RETURNS_DEEP_STUBS) LanguageServiceClient languageServiceMock;
-  @Mock(answer = Answers.RETURNS_DEEP_STUBS) Sentiment sentimentMock;
 
   @Spy HttpServletResponse responseSpy;
 
   @InjectMocks SentimentServlet sentimentServlet;
 
   @Before
-  public void setUp() throws IOException, Exception {
+  public void setUp() throws IOException {
+    when(requestMock.getParameter("video-id")).thenReturn(VIDEO_ID);
```

Comment:
Even though this is happening in every test, I believe this mock should be in each test for two reasons:

(1) Its the main input to the servlet you are testing. Each test should know what the input is for the reader.
(2) Your test mock other service requests that depend on what the input to the servlet is. For example, when you mock getCommentsFromId and getCaptionFromId you give the video id. Not obvious to the reader why that video id is or where it came from. You want the test to be read like "If the input to the servlet is VIDEO_ID and when getCommentsFromId gets VIDEO_ID then return blah and in the end the behavior blah."